cmd/job-executor-service/main.go

```diff
@@ -223,43 +192,60 @@ func _main(args []string, env envConfig) int {
 
 	log.Printf("Creating new http handler")
 
-	// configure http server to receive cloudevents
-	listeningAddr := fmt.Sprintf("localhost:%d", env.Port)
-	listener, err := net.Listen(
-		"tcp", listeningAddr,
-	)
+	imageFilterList, err := utils.BuildImageAllowList(env.AllowedImageList)
 	if err != nil {
-		log.Fatalf("error listening on tcp %s: %v", listeningAddr, err)
+		log.Fatalf("failed to generate the allowlist, %v", err)
 	}
 
-	p, err := cloudevents.NewHTTP(
-		cloudevents.WithPath(env.Path),
-		cloudevents.WithListener(
-			listener,
-		),
-	)
+	log.Fatal(sdk.NewKeptn(
+		serviceName,
+		sdk.WithTaskHandler(
+			eventWildcard,
+			NewEventHandler(imageFilterList),
+			jobEventFilter),
+		sdk.WithLogger(logrus.New()),
+	).Start())
+}
 
-	if err != nil {
-		log.Fatalf("failed to create client, %v", err)
+func jobEventFilter(keptnHandle sdk.IKeptn, event sdk.KeptnEvent) bool {
+	keptnHandle.Logger().Infof("Received event of type: %s from %s with id: %s", *event.Type, *event.Source, event.ID)
+
+	data := &keptnv2.EventData{}
+	if err := keptnv2.Decode(event.Data, data); err != nil {
+		keptnHandle.Logger().Errorf("Could not parse event: %s", err.Error())
+		return false
 	}
-	c, err := cloudevents.NewClient(p)
-	if err != nil {
-		log.Fatalf("failed to create client, %v", err)
+
+	jcr := &config.JobConfigReader{
+		Keptn: keptn_interface.NewV1ResourceHandler(*data, keptnHandle.APIV2().Resources()),
 	}
 
-	imageFilterList, err := utils.BuildImageAllowList(env.AllowedImageList)
+	// Check if the job configuration can be found
+	configuration, _, err := jcr.GetJobConfig(event.GitCommitID)
+
 	if err != nil {
-		log.Fatalf("failed to generate the allowlist, %v", err)
+		keptnHandle.Logger().Infof("could not retrieve config for job-executor-service: %e", err)
```

Comment:
We should send an error log event, which was previously done by the eventhandler itself, e.g.:
```
errorLogErr := eh.ErrorSender.SendErrorLogEvent(
			eh.Keptn.CloudEvent, fmt.Errorf(
				"could not retrieve config for job-executor-service: %w", err,
			),
		)
```

Reply:
Let's do this in a separate PR since this one is already quite large.